x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportStartTransformAction.java

```diff
@@ -430,7 +430,7 @@ public boolean test(PersistentTasksCustomMetaData.PersistentTask<?> persistentTa
                 // Consider this a failure.
                 exception = new ElasticsearchStatusException(
                     "Could not start transform, allocation explanation [" + assignment.getExplanation() + "]",
-                    RestStatus.TOO_MANY_REQUESTS
+                    RestStatus.CONFLICT
```

Comment:
Since we currently don't limit the number of transforms, this change makes sense. 

If we ever add a setting that limits how many transforms can run on a node, we should revisit this.

Reply:
Agree, when looking at it, wondered if we can add a status code to the assignment class to avoid having `if`s to determine the response code.

Reply:
I think @droberts195 has thought of something similar.

That would also help us in situations where we want to wait for a different assignment. Right now it is a fairly brittle substring checking system .

Reply:
I agree checking for various strings is brittle and it would be nicer to have codes we could check for, but that's probably a complex change in the persistent tasks framework (considering BWC) so it can be decoupled from this PR.

One thing that _would_ be nice to add to this PR is a special message that mentions the exact name of the `cluster.persistent_tasks.allocation.enable` setting when that setting is the cause of the assignment failure.  https://github.com/elastic/elasticsearch/issues/52826#issuecomment-591500744 refers to this and the change that added it was #52014 (albeit with the wrong response code - that needs changing still).